include/ion/building_block.h

```diff
@@ -28,6 +28,13 @@ using Input = Halide::GeneratorInput<T>;
 template<typename T>
 using Output = Halide::GeneratorOutput<T>;
 
+template<typename T>
+using DynamicInput = Halide::GeneratorInput<T>;
```

Comment:
The type alias is not necessary for this purpose. You can simply use Input or Output